modules/packages/LinearAlgebra.chpl

```diff
@@ -221,6 +222,267 @@ class LinearAlgebraError : Error {
     }
 }
 
+/*
+This class promotes lazy computation of required
+Helper matrices to evaluate matrix exponential.
+
+High precision of expm requires smaller computations,
+hence we do a lazy computation depending on the norm value
+of the matrix.
+
+*/
+class ExpmPadeHelper {
+  type t;
+  var A : t;
+  var A2 : t;
+  var A4 : t;
+  var A6 : t;
+  var A8 : t;
+  var A10 : t;
+  var Exactd4 : real;
+  var Exactd6 : real;
+  var Exactd8 : real;
+  var Exactd10 : real;
+  var Approxd4 : real;
+  var Approxd6 : real;
+  var Approxd8 : real;
+  var Approxd10 : real;
+  var ident : t;
+  var use_exact_onenorm : bool;
+  // Boolean arrays to know if A, Exactd, Approxd
+  // have been computed.
+  var isAComputed : [1..5] bool;
+  var isExactdComputed : [1..4] bool;
+  var isApproxdComputed : [1..4] bool;
```

Comment:
These seem like they could make use of associative arrays, where the index value is more descriptive of what they represent (see [the associative array primer](https://chapel-lang.org/docs/latest/primers/associative.html) for examples).  You could also instead use [sparse](https://chapel-lang.org/docs/latest/primers/sparse.html) arrays or arrays created over [strided](https://chapel-lang.org/docs/latest/primers/ranges.html#operators) indices so that the index corresponds with the field name more naturally